[feature](cache) support file cache admission control

wenzhenghu opened the pull request:
To fully understand the implementation of the PR, please refer to the following link（It is a Chinese document）: https://www.notion.so/V3-1-2c31293e1081807ca476dd5c87efb28e

### 1. PR Function Overview

The core function of this PR is the implementation of **File Cache Admission Control**.

*   **Background**: Doris supports using File Cache to accelerate data access for remote storage (e.g., HDFS, S3). However, in certain scenarios (such as large-scale ETL jobs or heavy ad-hoc queries), reading massive amounts of cold data can evict existing hot data from the File Cache. This leads to "cache pollution" and a significant drop in cache hit rates.
*   **Goal**: Provide a mechanism to decide whether data scanned by a specific query is allowed to enter the File Cache, based on dimensions such as user identity and table information. If admission is denied, the query will read the data directly from remote storage without populating the cache.

### 2. Implementation Scheme Analysis

The implementation consists of the following key components:

#### 2.1. FE Side: Admission Decision

The primary logic is located in the `createScanRangeLocations` method of `FileQueryScanNode.java`.

*   **New Configuration**: Introduced the `Config.enable_file_cache_admission_control` switch.
*   **New Manager**: Introduced `FileCacheAdmissionManager` (Singleton) to execute the specific admission judgment logic.
*   **Decision Flow**:
    1.  Before generating Scan Ranges, `FileQueryScanNode` retrieves the current User Identity (`userIdentity`), Catalog, Database, and Table information.
    2.  It calls `FileCacheAdmissionManager.getInstance().isAllowed(...)` to obtain a boolean result `fileCacheAdmission`.
    3.  It logs the decision result and the time cost in query plan.
```
|   0:VHIVE_SCAN_NODE(74)                                                                                       |
|      table: test_file_cache_features.tpch1_parquet.lineitem                                                   |
|      inputSplitNum=10, totalFileSize=205792918, scanRanges=10                                                 |
|      partition=1/1                                                                                            |
|      cardinality=1469949, numNodes=1                                                                          |
|      pushdown agg=NONE                                                                                        |
|      file cache request ADMITTED: user_identity:root@%, reason:user table-level whitelist rule, cost:37996 ns |
|      limit: 1 
```

#### 2.2. FE Side: Decision Propagation

The decision result needs to be propagated from the `FileQueryScanNode` down to the underlying split assignment logic.

*   **SplitAssignment Modification**:
    *   The constructor of the `SplitAssignment` class (located in `org.apache.doris.datasource`) is modified to accept a new `boolean fileCacheAdmission` parameter.

*   **SplitToScanRange Modification**:
    *   The `splitToScanRange` method (or its corresponding Lambda expression) is updated to receive the `fileCacheAdmission` parameter.
    *   This method is responsible for setting this value into the Thrift object.

#### 2.3. Communication Protocol: Thrift Definition Update

To pass the FE's decision to the BE, the Thrift definition (likely `TFileRangeDesc` or `TFileScanRangeParams` in `PlanNodes.thrift`) requires a new field.

*   **Inferred Change**: A new field `optional bool file_cache_admission`, is added to the `TFileRangeDesc` struct.

#### 2.4. BE Side: Enforcement

Although the analysis focuses on the FE, the complete loop requires enforcement on the BE side:

*   **FileReader**: The BE's `FileReader` (e.g., `HdfsFileReader` or `S3FileReader`) checks the `file_cache_admission` flag in the incoming `TFileRangeDesc` during initialization or reading.
*   **Cache Policy**:
    *   If `file_cache_admission` is **true** (default): It uses the standard `FileCachePolicy`, where data not found in the cache is written to the Block File Cache after reading.
    *   If `file_cache_admission` is **false**: It sets the `FileCachePolicy` to `NO_CACHE`, skips the cache writing step, reading directly from remote storage. This protects the existing cache from being polluted.

### 3. Summary

This PR introduces an **Admission Control Manager** during the FE query planning phase and transparently passes this control signal through the **Split Assignment** and **Scan Range Generation** phases. This ultimately guides the BE side's file readers to **selectively** use the file cache. 

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [x] Unit Test
    - [x] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [ ] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [ ] No.
    - [x] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

xuchenhao committed:
Test feasibility.

xuchenhao committed:
Implement the basic functionality.

xuchenhao committed:
Add some descriptions.

xuchenhao committed:
Add FileCacheAdmissionManager.java.

xuchenhao committed:
Optimize code logic.

xuchenhao committed:
Update the order of rule matching.

xuchenhao committed:
Update the default value of file_cache_admission.

xuchenhao committed:
Update the level of LOG.

xuchenhao committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into file_cache_admission_control

xuchenhao committed:
Updat the rule matching order.

xuchenhao committed:
Add ut cases.

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  55.99% (229/409)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_613dab599e351a5d89f129bb2435e6940dc39fd2/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_613dab599e351a5d89f129bb2435e6940dc39fd2/fe_report/index.html)

xuchenhao committed:
Update the measurement of cost.

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  55.99% (229/409)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_12de8571c1c59eb699be1e119906ce440888e455/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_12de8571c1c59eb699be1e119906ce440888e455/fe_report/index.html)

xuchenhao committed:
Update pom.xml

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `60.00% (3/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/bb5c8865436513edd83e1e4676ce03ac98e8b9b6_bb5c8865436513edd83e1e4676ce03ac98e8b9b6/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/bb5c8865436513edd83e1e4676ce03ac98e8b9b6_bb5c8865436513edd83e1e4676ce03ac98e8b9b6/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 53.39% (18824/35256) |
| Line Coverage     | 39.20% (174357/444843) |
| Region Coverage   | 33.77% (134981/399701) |
| Branch Coverage   | 34.67% (58130/167681) |

xuchenhao committed:
Store admission rules in JSON files.

xuchenhao committed:
Update FileCacheAdmissionManager.

xuchenhao commented:
run buildall.

xuchenhao committed:
Update export_mysql_rule_to_json.sh.

xuchenhao committed:
Add license.

xuchenhao commented:
run buildall.

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  57.77% (212/367)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_d49d83331415976251cfe2699cb43dc91cb7e5b3/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_d49d83331415976251cfe2699cb43dc91cb7e5b3/fe_report/index.html)

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  57.77% (212/367)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_17f688e84ebec45c2614235cf87cbe7861b029ac/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_17f688e84ebec45c2614235cf87cbe7861b029ac/fe_report/index.html)

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `60.00% (3/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/17f688e84ebec45c2614235cf87cbe7861b029ac_17f688e84ebec45c2614235cf87cbe7861b029ac/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/17f688e84ebec45c2614235cf87cbe7861b029ac_17f688e84ebec45c2614235cf87cbe7861b029ac/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 53.38% (18944/35488) |
| Line Coverage     | 39.24% (175624/447562) |
| Region Coverage   | 33.81% (135863/401787) |
| Branch Coverage   | 34.72% (58622/168863) |

xuchenhao committed:
Update FileCacheAdmissionManager.java

xuchenhao commented:
run buildall

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  57.77% (212/367)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_9bdf1119e521da020134c7d06c14c53815c75625/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_9bdf1119e521da020134c7d06c14c53815c75625/fe_report/index.html)

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `60.00% (3/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/9bdf1119e521da020134c7d06c14c53815c75625_9bdf1119e521da020134c7d06c14c53815c75625/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/9bdf1119e521da020134c7d06c14c53815c75625_9bdf1119e521da020134c7d06c14c53815c75625/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 53.38% (18944/35488) |
| Line Coverage     | 39.23% (175594/447562) |
| Region Coverage   | 33.79% (135744/401787) |
| Branch Coverage   | 34.70% (58593/168863) |

hello-stephen commented:
# BE Regression && UT Coverage Report
Increment line coverage `100.00% (5/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_9bdf1119e521da020134c7d06c14c53815c75625_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_9bdf1119e521da020134c7d06c14c53815c75625_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 72.24% (25063/34693) |
| Line Coverage     | 58.94% (263111/446395) |
| Region Coverage   | 53.79% (218368/405990) |
| Branch Coverage   | 55.37% (93814/169443) |

xuchenhao committed:
Update pom.xml

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Update FileCacheAdmissionManager.java

xuchenhao commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/4a959e6a4f04e72d46d880dabd383856bf13f944_4a959e6a4f04e72d46d880dabd383856bf13f944_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/4a959e6a4f04e72d46d880dabd383856bf13f944_4a959e6a4f04e72d46d880dabd383856bf13f944_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.52% (1771/2227) |
| Line Coverage     | 64.83% (31290/48262) |
| Region Coverage   | 65.36% (15560/23808) |
| Branch Coverage   | 56.01% (8279/14780) |

xuchenhao committed:
Update export_mysql_rule_to_json.sh

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/836850619a968afda07cdefa0b1ac6baee778c50_836850619a968afda07cdefa0b1ac6baee778c50_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/836850619a968afda07cdefa0b1ac6baee778c50_836850619a968afda07cdefa0b1ac6baee778c50_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.52% (1771/2227) |
| Line Coverage     | 64.84% (31294/48262) |
| Region Coverage   | 65.40% (15571/23808) |
| Branch Coverage   | 56.00% (8277/14780) |

hello-stephen commented:
# BE Regression && UT Coverage Report
Increment line coverage `100.00% (5/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_836850619a968afda07cdefa0b1ac6baee778c50_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_836850619a968afda07cdefa0b1ac6baee778c50_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 72.19% (25044/34694) |
| Line Coverage     | 58.85% (262764/446486) |
| Region Coverage   | 53.53% (217430/406174) |
| Branch Coverage   | 55.18% (93529/169504) |

hello-stephen commented:
# FE Regression Coverage Report
Increment line coverage `  1.09% (4/367)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_836850619a968afda07cdefa0b1ac6baee778c50_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_836850619a968afda07cdefa0b1ac6baee778c50_merge_fe/report/index.html)

xuchenhao committed:
Enhanced compatibility.

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/a795d77a3f6e4c9d894f6c0eef8095bcabeab072_a795d77a3f6e4c9d894f6c0eef8095bcabeab072_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/a795d77a3f6e4c9d894f6c0eef8095bcabeab072_a795d77a3f6e4c9d894f6c0eef8095bcabeab072_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.57% (1780/2237) |
| Line Coverage     | 64.87% (31543/48624) |
| Region Coverage   | 65.40% (15686/23986) |
| Branch Coverage   | 56.00% (8336/14886) |

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# FE Regression Coverage Report
Increment line coverage `  1.27% (5/395)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_65665f9ff20c463d255f9f5ce139ef80b023df37_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_65665f9ff20c463d255f9f5ce139ef80b023df37_merge_fe/report/index.html)

xuchenhao committed:
Refactored variable names.

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/2fdb76489ab0cb878d535e9abf3bf200e336782a_2fdb76489ab0cb878d535e9abf3bf200e336782a_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/2fdb76489ab0cb878d535e9abf3bf200e336782a_2fdb76489ab0cb878d535e9abf3bf200e336782a_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.53% (1779/2237) |
| Line Coverage     | 64.71% (31476/48643) |
| Region Coverage   | 65.40% (15659/23943) |
| Branch Coverage   | 55.98% (8316/14856) |

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Refactor variable names.

xuchenhao commented:
run buildall

morningman commented on fe/fe-core/src/main/java/org/apache/doris/datasource/FileCacheAdmissionManager.java:
```diff
@@ -0,0 +1,697 @@
+        }
+    }
+
+    public static class ConcurrentRuleCollection {
```
What is `concurrent` mean here?

morningman commented on fe/fe-core/src/main/java/org/apache/doris/datasource/FileCacheAdmissionManager.java:
```diff
@@ -0,0 +1,697 @@
+        private final Map<String, Set<String>> includeDatabaseRules = new HashMap<>();
+        private final Map<String, Set<String>> includeTableRules = new HashMap<>();
+
+        static List<String> reasons = new ArrayList<>(Arrays.asList(
```
How about just define each reason as `public static final String`?
Because it is hard to maintain when using array index

xuchenhao replied:
ok

morningman commented on fe/fe-core/src/main/java/org/apache/doris/datasource/FileQueryScanNode.java:
```diff
@@ -307,11 +309,42 @@ public void createScanRangeLocations() throws UserException {
+                long endTime = System.nanoTime();
+                double durationMs = (double) (endTime - startTime) / 1_000_000;
+
+                LOG.debug("File cache admission control cost {} ms", String.format("%.6f", durationMs));
```
if (LOG.isDebugEnable())

morningman reviewed (commented):
BTW, could you give a complete rule json example?
And does it easy to extend to internal table?

wenzhenghu commented on fe/fe-core/src/main/java/org/apache/doris/datasource/FileCacheAdmissionManager.java:
```diff
@@ -0,0 +1,697 @@
+        }
+    }
+
+    private void startRefreshTask() {
```
ConfigWatcher is better

xuchenhao committed:
Address review comments and refine code.

wenzhenghu commented:
> BTW, could you give a complete rule json example? And does it easy to extend to internal table?
1. example rule json file：
```
[
  {
    "id": 395744,
    "enabled": 1,
    "rule_type": 0,
    "table_name": "",
    "catalog_name": "noise_catalog_42",
    "created_time": 1766557246,
    "updated_time": 1766557246,
    "database_name": "noise_db_46",
    "user_identity": "Mgywiqjyt@%",
    "partition_pattern": ""
  },
  {
    "id": 395745,
    "enabled": 1,
    "rule_type": 0,
    "table_name": "noise_table_74",
    "catalog_name": "noise_catalog_50",
    "created_time": 1766557246,
    "updated_time": 1766557246,
    "database_name": "noise_db_85",
    "user_identity": "Chaykxshz@%",
    "partition_pattern": ""
  },
  {
    "id": 395746,
    "enabled": 1,
    "rule_type": 0,
    "table_name": "noise_table_14",
    "catalog_name": "noise_catalog_28",
    "created_time": 1766557246,
    "updated_time": 1766557246,
    "database_name": "noise_db_11",
    "user_identity": "yvnuuocpv@%",
    "partition_pattern": ""
  }
]
```
2. it could extend to internal table scenario. If needed, we can implement internal table admission control in subsequent PRs.

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/ae4a01138cabae4fbf1f60a62afd120991167f61_ae4a01138cabae4fbf1f60a62afd120991167f61_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/ae4a01138cabae4fbf1f60a62afd120991167f61_ae4a01138cabae4fbf1f60a62afd120991167f61_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.57% (1784/2242) |
| Line Coverage     | 64.76% (31732/48997) |
| Region Coverage   | 65.41% (15776/24120) |
| Branch Coverage   | 55.99% (8381/14968) |

xuchenhao commented:
run buildall

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Address review comments and refine code.

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  59.95% (259/432)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_25a419eb9a00c5a83817c54d8c901f9634b53fe3/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_25a419eb9a00c5a83817c54d8c901f9634b53fe3/fe_report/index.html)

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/012257c158fab77cfcdfd647e829553669497b17_012257c158fab77cfcdfd647e829553669497b17_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/012257c158fab77cfcdfd647e829553669497b17_012257c158fab77cfcdfd647e829553669497b17_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.52% (1786/2246) |
| Line Coverage     | 64.75% (31802/49114) |
| Region Coverage   | 65.46% (15842/24201) |
| Branch Coverage   | 55.98% (8413/15028) |

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# BE Regression && UT Coverage Report
Increment line coverage `100.00% (5/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_0f6ea1825fecb7ade4922f639916d6785bef3c39_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_0f6ea1825fecb7ade4922f639916d6785bef3c39_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 71.44% (25446/35619) |
| Line Coverage     | 53.95% (266209/493472) |
| Region Coverage   | 51.55% (220703/428117) |
| Branch Coverage   | 52.95% (94740/178921) |

wenzhenghu committed:
Update descriptions for file cache admission control config

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

hello-stephen commented:
# BE Regression && UT Coverage Report
Increment line coverage `100.00% (5/5)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_f1753dda6ad16df2391ff4e2d29589bd13ed0622_merge/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_f1753dda6ad16df2391ff4e2d29589bd13ed0622_merge/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 71.48% (25592/35802) |
| Line Coverage     | 54.11% (267566/494447) |
| Region Coverage   | 51.70% (222638/430669) |
| Branch Coverage   | 53.18% (95748/180059) |

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  59.95% (259/432)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_d8990a578b9672af399f6bcf5171e77dbf9fee72/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_d8990a578b9672af399f6bcf5171e77dbf9fee72/fe_report/index.html)

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/0293a49b09700b00d6e7aa9da5af94be840ac965_0293a49b09700b00d6e7aa9da5af94be840ac965_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/0293a49b09700b00d6e7aa9da5af94be840ac965_0293a49b09700b00d6e7aa9da5af94be840ac965_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.29% (1792/2260) |
| Line Coverage     | 64.61% (31868/49323) |
| Region Coverage   | 65.37% (15892/24309) |
| Branch Coverage   | 55.91% (8443/15100) |

xuchenhao committed:
[feature](file-cache) implement file cache admission control

This commit implements file cache admission control, which allows users to control which files/tables/partitions should be cached in the file cache.

Key features:
- FileCacheAdmissionManager for managing admission rules.
- Support for rule matching at global, catalog, database, and table levels.
- Rules can be stored in JSON files and auto-refreshed.
- Integration with FileQueryScanNode for admission check during query execution.
- Added unit tests for FileCacheAdmissionManager.

xuchenhao committed:
Merge remote-tracking branch 'origin/file_cache_admission_control' into file_cache_admission_control

xuchenhao committed:
Merge upstream/master and handle readers rename to be/src/format

xuchenhao committed:
refactor: address review comments for file cache admission control

- BE: Add Thrift __isset checks for file_cache_admission in various readers
- FE: Fix lock safety with try-finally blocks in FileCacheAdmissionManager
- FE: Add null check for File.listFiles() to prevent NPE
- FE: Replace Character.isAlphabetic with strict ASCII range check
- FE: Rename ConcurrentRuleManager to RuleManager for clarity
- FE: Refactor Boolean to primitive boolean to avoid autoboxing
- Test: Replace Thread.sleep with Awaitility in FileCacheAdmissionRuleRefresherTest

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao commented on be/src/format/csv/csv_reader.cpp:
```diff
@@ -228,6 +228,7 @@ void CsvReader::_init_file_description() {
         _file_description.fs_name = _range.fs_name;
     }
+    _file_description.file_cache_admission = _range.file_cache_admission;
 }
```
Done

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
[feature](cache) address review comments for file cache admission

xuchenhao committed:
Merge remote-tracking branch 'upstream/master' into file_cache_admission_control

xuchenhao committed:
Merge remote-tracking branch 'origin/file_cache_admission_control' into file_cache_admission_control

xuchenhao commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  58.20% (259/445)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/59065_22d31cdbf0a989ea272784b05dee605d2f6b0747/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/59065_22d31cdbf0a989ea272784b05dee605d2f6b0747/fe_report/index.html)

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control

morningman committed:
remove chinese description

morningman commented:
run buildall

hello-stephen commented:
# Cloud UT Coverage Report
Increment line coverage ` ` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/fea3cd22836138d70cd1102c48f3687b00f3aab9_fea3cd22836138d70cd1102c48f3687b00f3aab9_cloud/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/fea3cd22836138d70cd1102c48f3687b00f3aab9_fea3cd22836138d70cd1102c48f3687b00f3aab9_cloud/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 79.15% (1796/2269) |
| Line Coverage     | 64.42% (32257/50072) |
| Region Coverage   | 65.35% (16160/24727) |
| Branch Coverage   | 55.77% (8607/15434) |

wenzhenghu committed:
Merge branch 'master' into file_cache_admission_control

wenzhenghu committed:
Merge branch 'master' into file_cache_admission_control

wenzhenghu commented:
run buildall

xuchenhao committed:
Merge branch 'master' into file_cache_admission_control
